### ISSUE TITLE
SA0102 : False Positive (CSharp.CsParser : A syntax error has been discovered)

### ISSUE THREAD

**Issue by Roman-Didenko** [Visual-Stylecop/Visual-StyleCop#51]
The following code:

```
Func<object> x = () => null;
decimal t = x() as decimal? ?? 0;
```

incorrectly gives:
`SA0102 : CSharp.CsParser : A syntax error has been discovered in file ....`

Build 4.7.58.3

**Comment by carl-berg** [comment]
I have also discovered a false positive for this rule. This one regarding C#6 strings:
```
var from = "2015-01-01";
var to = 2016-01-01";
var token = $"{{ \"From\": \"{from}\", \"To\": \"{to}\" }}";
```

**Comment by pbalachandar** [comment]
I get SA0102 : CSharp.CsParser : A syntax error has been discovered in file... for the below line

public static MySingletonInstance Instance => LazyInstance.Value;

How can I fix this?

**Comment by onegronm** [comment]
I have the same false positive with this line of code: 

``` csharp
dataEntryOutlineDiv.Attributes["class"] = "divHeightButton";
```

**Comment by rsheptolut** [comment]
The false positive is not as big a problem as it is the fact that I get no style checking for the file at all. So it is in fact a failure of StyleCop
